### PR TITLE
Fix CMake warnings and errors

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@rdumusc FYI

File: DoxygenRule.cmake [hunk 1]
```diff
@@ -141,7 +141,7 @@ set(_jekyll_md_file "${PROJECT_BINARY_DIR}/doc/${PROJECT_NAME}-${VERSION_MAJOR}.
 file(WRITE ${_jekyll_md_file}
 "---\n"
 "name: ${PROJECT_NAME}\n"
-"version: "${VERSION_MAJOR}.${VERSION_MINOR}"\n"
+"version: \"${VERSION_MAJOR}.${VERSION_MINOR}\"\n"
```
Comment (rdumusc) [inline DoxygenRule.cmake]: ooops sorry for that..!

**Comment by rdumusc** [comment]
+1

File: DoxygenRule.cmake (hunk 1 above)
Comment (eile) [inline DoxygenRule.cmake]: No worries. I also had one like this.

Comment (tribal-tec) [inline DoxygenRule.cmake]: Good. I just wrote to @rdumusc about this :)

**Pull request merged**